src/main/java/io/dropwizard/kafka/security/SaslPlaintextSecurityFactory.java

```diff
@@ -0,0 +1,41 @@
+package io.dropwizard.kafka.security;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.collect.ImmutableMap;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import java.util.Map;
+import java.util.Optional;
+
+@JsonTypeName("sasl_plaintext")
+public class SaslPlaintextSecurityFactory extends SecurityFactory {
+
+    @NotEmpty
+    @JsonProperty
+    private String saslMechanism = "PLAIN";
+
+    @JsonProperty
+    private Optional<String> saslJaas = Optional.empty();
```

Comment:
The ```dropwizard-configuration``` module allows one to inject properties through env vars. Actually, we are utilising the same.

If the property is neither configured nor injected, authorisation exception will be thrown. @mzamani1 what's your take on this?

Reply:
Hm, this may have been an oversight in my review, but if the saslJaas is truly required, or else an authorization exception is thrown, then I think it should be declared as not optional.